### DISCUSSION TITLE
PODS_DISABLE_CONTENT_MENU adverse effect

### DISCUSSION THREAD

**Discussion by EthanPhoenix** [pods-framework/pods#6767] [Ideas]
## Issue Overview
Using: `define( 'PODS_DISABLE_CONTENT_MENU', true );` or `add_filter( 'PODS_DISABLE_CONTENT_MENU', '__return_false' );` and unchecking "Show Admin Menu in Dashboard" causes a “Sorry, you are not allowed to access this page.” error when accessing the Pod’s admin UI form (also via an “Add New” modal).

See this [post](https://pods.io/forums/topic/removing-all-pods-menu-entries-from-the-admin-menu/#post-198702) for more info.

## Expected Behavior
The code should simply remove the additional "Pods" menu that is being created and not restrict the access to the pods.
In fact, the more extended expected behavior is that once "Show Admin Menu in Dashboard" is unchecked there shouldn't be any need to add any code to remove a secondary menu that is generated as a result of that (I believe it beats the purpose of what the checkbox is suppose to be doing).

## Current Behavior
Using one of the aforementioned codes combined with unchecking "Show Admin Menu in Dashboard" results in “Sorry, you are not allowed to access this page.”

## Possible Solution
1. Using any one of the codes with `PODS_DISABLE_CONTENT_MENU` should only remove the menu entry but not the rights to access the pods' UI.
2 . When unchecking "Show Admin Menu in Dashboard" don't generate any kind of menu entry and leave the possibility to access the UI via an "Add New" or a self created link.

## Steps to Reproduce (for bugs)
1. Create two pods.
2. Make one relate to another.
3. Make sure to uncheck the "Show Admin Menu in Dashboard" under the Admin UI tab in both pods.
4. While the menu entries for both pods disappear from the first level, they suddently appear in a new menu entry called 'Pods'.
5. Add one of the codes above in php. This will remove the new menu entry.
6. Try to access one of the pods using the URL (...asmin.php?page=pods-manage-{pods_name}) you'll get a permission error.
7. "Bonus": leave one of the the related pods "Show Admin Menu in Dashboard" checkbox checked. Add new within it and in the relationship field to the other pod try to "Add New" - same permission error.

## Pods Package Export (helpful!)

```
{"meta":{"version":"2.7.3","build":1526512023},"pods":{"4":{"id":4,"name":"test_a_db","label":"Test A","description":"","type":"pod","storage":"table","object":"","alias":"","fields":{"name":{"id":5,"name":"name","label":"Name","description":"","help":"","class":"","type":"text","weight":0,"pick_object":"","pick_val":"","sister_id":"","required":"1","unique":"0","text_repeatable":"0","text_allowed_html_tags":"strong em a ul ol li b i","text_max_length":"255","text_placeholder":""},"relationship_field":{"id":10,"name":"relationship_field","label":"Relationship Field","description":"","help":"","class":"","type":"pick","weight":1,"pick_object":"pod","pick_val":"test_b_db","sister_id":"","required":"0","unique":"0","pick_format_type":"single","pick_format_single":"dropdown","pick_format_multi":"checkbox","pick_allow_add_new":"1","pick_taggable":"0","pick_show_icon":"1","pick_show_edit_link":"1","pick_show_view_link":"1","pick_limit":"0","pick_user_role":[],"pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"pick_select_text":"","pick_table_id":"","pick_table_index":"","pick_display":"","pick_where":"","pick_orderby":"","pick_groupby":""}},"show_in_menu":"0","label_singular":"Test A","pod_index":"name","hierarchical":"0","ui_style":"post_type","ui_actions_enabled":["add","edit","duplicate","delete"],"ui_reorder_field":"menu_order","ui_fields_manage":["name"]},"11":{"id":11,"name":"test_b_db","label":"Test B","description":"","type":"pod","storage":"table","object":"","alias":"","fields":{"name":{"id":12,"name":"name","label":"Name","description":"","help":"","class":"","type":"text","weight":0,"pick_object":"","pick_val":"","sister_id":"","required":"1","unique":"0","text_repeatable":"0","text_allowed_html_tags":"strong em a ul ol li b i","text_max_length":"255","text_placeholder":""}},"show_in_menu":"0","label_singular":"Test B","pod_index":"name","hierarchical":"0","ui_style":"post_type","ui_actions_enabled":["add","edit","duplicate","delete"],"ui_reorder_field":"menu_order","ui_fields_manage":{"0":"name","2":"modified"}}}}
```

## Todos
- [ ] Tests
- [ ] Documentation

## WordPress Environment
<details>
```
WordPress Version: 4.9.5

PHP Version: 7.0.28

MySQL Version: 5.7.21

Server Software: Apache

Your User Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36 OPR/52.0.2871.99

Session Save Path: C:/Bitnami/wordpress-4.9.5-0/php/tmp

Session Save Path Exists: Yes

Session Save Path Writeable: Yes

Session Max Lifetime: 1440

Opcode Cache:

Apc: No
Memcached: No
OPcache: Yes
Redis: No
Object Cache:

APC: No
APCu: No
Memcache: No
Memcached: No
Redis: No
WPDB Prefix: wp_

WP Multisite Mode: No

WP Memory Limit: 40M

Pods Network-Wide Activated: No

Pods Install Location: C:\Bitnami\wordpress-4.9.5-0\apps\wordpress\htdocs\wp-content\plugins\pods/

Pods Tableless Mode Activated: No

Pods Light Mode Activated: No

Currently Active Theme: Twenty Seventeen

Currently Active Plugins:

Ethan's Pods Addon: 0.0.1
Pods - Custom Content Types and Fields: 2.7.3```
</details>

**Comment by quasel** [comment]
is that something to look at @sc0ttkclark? never been in there :D ^^
